### PR TITLE
Add: JWT validation and GKE troubleshooting

### PR DESCRIPTION
**Pull request by MilanBariya**
### Describe your changes :

I worked on the adding JWT validation troubleshooting page.
Fix: #8602 

#
### Type of change :

- [x] Documentation

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-docs/content/deployment/kubernetes/gke-troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  src="/images/deployment/troubleshoot/permission-pod-events.png"
+  alt="permission-pod-events"
+  caption="Permission pod events"
+/>
+</div>
+
+Please validate all the prerequisites mentioned in this [section](/deployment/kubernetes/gke).
+
+Please validate the configuration of `dags_pv_pvc.yml` file.
+Please validate `storageClassName` filed in YAML file.
```
Comment (nahuelverdugo) [inline openmetadata-docs/content/deployment/kubernetes/gke-troubleshooting.md:28]:
```suggestion
Please validate:
- all the prerequisites mentioned in this [section](/deployment/kubernetes/gke)
- the configuration of `dags_pv_pvc.yml` file
- `storageClassName` filed in YAML file
```

File: openmetadata-docs/content/deployment/kubernetes/gke-troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+---
+title: GKE Setup Troubleshooting
+slug: /deployment/kubernetes/gke-troubleshooting
+---
+
+# GKE Setup Troubleshooting
+
+If you came across `invalid access type while creating the pvc. and the permission pod is stuck at "pending" state`  
+The above error might have occurred due to the pvc volumes not setup or pvc volumes are not mount properly
```
Comment (nahuelverdugo) [inline openmetadata-docs/content/deployment/kubernetes/gke-troubleshooting.md:9]:
```suggestion

The above error might have occurred due to the pvc volumes not setup or pvc volumes are not mounted properly.
```

File: openmetadata-docs/content/deployment/kubernetes/gke-troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+---
+title: GKE Setup Troubleshooting
+slug: /deployment/kubernetes/gke-troubleshooting
+---
+
+# GKE Setup Troubleshooting
+
+If you came across `invalid access type while creating the pvc. and the permission pod is stuck at "pending" state`
```
Comment (nahuelverdugo) [inline openmetadata-docs/content/deployment/kubernetes/gke-troubleshooting.md:8]:
```suggestion
If you came across `invalid access type while creating the pvc`, and the permission pod is stuck in "pending" state.
```

File: openmetadata-docs/content/deployment/kubernetes/gke-troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+# GKE Setup Troubleshooting
+
+If you came across `invalid access type while creating the pvc. and the permission pod is stuck at "pending" state`  
+The above error might have occurred due to the pvc volumes not setup or pvc volumes are not mount properly
+
+<div className="w-100 flex justify-center">
+<Image
+  src="/images/deployment/troubleshoot/dag-log.png"
+  alt="dag-log"
```
Comment (nahuelverdugo) [inline openmetadata-docs/content/deployment/kubernetes/gke-troubleshooting.md:14]: "dag-log" will be the footer of the image, is it correct?

File: openmetadata-docs/content/deployment/security/jwt-troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+---
+title: JWT validation Troubleshooting
+slug: /deployment/security/jwt-troubleshooting
+---
+# JWT Troubleshooting
+
+Add the `{domain}:{port}/config/jwks` in the list of publicKeyUrls
```
Comment (nahuelverdugo) [inline openmetadata-docs/content/deployment/security/jwt-troubleshooting.md:7]:
```suggestion
Add the `{domain}:{port}/config/jwks` in the list of `publicKeyUrls`:
```

Comment (nahuelverdugo) [reply openmetadata-docs/content/deployment/security/jwt-troubleshooting.md:7]: `publicKeyUrls` or `publicKeys`?

File: openmetadata-docs/content/deployment/security/jwt-troubleshooting.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+```yaml
+  authentication:
+    provider: "google"
+    publicKeys:
+    - "https://www.googleapis.com/oauth2/v3/certs"
+    - "http://localhost:8585/api/v1/config/jwks" (your domain and port)
+```
+
+This config with `"http://localhost:8585/api/v1/config/jwks"` is default behavior. if you configuring and expecting a jwt token to work, its important that config with that.
```
Comment (nahuelverdugo) [inline openmetadata-docs/content/deployment/security/jwt-troubleshooting.md:17]:
```suggestion
This config with `"http://localhost:8585/api/v1/config/jwks"` is the default behavior. If you are configuring and expecting a JWT token to work, configuring with that extra URL is required.
```

File: openmetadata-docs/content/deployment/security/jwt-troubleshooting.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+JWT Tokens are issued by private certificates.
+
+We need public keys to decrypt it and get at the user name and expiry time etc from that token.
```
Comment (nahuelverdugo) [inline openmetadata-docs/content/deployment/security/jwt-troubleshooting.md:21]:
```suggestion
We need public keys to decrypt it and get that token's user name, expiry time, etc.
```

File: openmetadata-docs/content/deployment/security/jwt-troubleshooting.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+In OpenMetadata users can enable SSO for users to login and use JWT tokens issued by OpenMetadata for bots
+The way OpenMetadata issues a JWT Token is using this [config](https://github.com/open-metadata/OpenMetadata/blob/main/conf/openmetadata.yaml#L155). It uses the rsaprivateKeyPath file to generate a token.
```
Comment (nahuelverdugo) [inline openmetadata-docs/content/deployment/security/jwt-troubleshooting.md:24]:
```suggestion
The way OpenMetadata issues a JWT Token is using this [config](https://github.com/open-metadata/OpenMetadata/blob/main/conf/openmetadata.yaml#L155). It uses the `rsapublicKeyFilePath` file to generate a token.
```

File: openmetadata-docs/content/deployment/security/jwt-troubleshooting.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+When the ingestion worfkflow uses this token , we use rsapublicKeyPath to decrypt it. The way we do this is using the response from this endpoint `http://localhost:8585/api/v1/config/jwks`.
```
Comment (nahuelverdugo) [inline openmetadata-docs/content/deployment/security/jwt-troubleshooting.md:26]:
```suggestion
When the ingestion workflow uses this token, we use `rsapublicKeyPath` to decrypt it. The way we do this is using the response from this endpoint `http://localhost:8585/api/v1/config/jwks`.
```

File: openmetadata-docs/content/deployment/security/jwt-troubleshooting.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+
+## Get JWT token from UI.
+
+First Open Open-Metadata UI than go to settings > Bots > Ingestion Bot
+
+<div className="w-100 flex justify-center">
+<Image
+  src="/images/deployment/troubleshoot/jwt-token.png"
+  alt="jwt-token"
```
Comment (nahuelverdugo) [inline openmetadata-docs/content/deployment/security/jwt-troubleshooting.md:36]: "jwt-token" will be the footer of the image, is it correct?

File: openmetadata-docs/content/deployment/security/jwt-troubleshooting.md (hunk 5 above)
Comment (MilanBariya) [reply openmetadata-docs/content/deployment/security/jwt-troubleshooting.md:7]: It should be `publicKeys`.

File: openmetadata-docs/content/deployment/security/jwt-troubleshooting.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+  caption="jwt-token"
```
[the remark this answers is not in the document]
Comment (MilanBariya) [reply openmetadata-docs/content/deployment/security/jwt-troubleshooting.md:37]: Yes it's correct

File: openmetadata-docs/content/deployment/security/jwt-troubleshooting.md [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+/>
+</div>
+
+You can validate that in [jwt.io](https://jwt.io/). if there's something wrong on how the JWT token was generated.
+
+<div className="w-100 flex justify-center">
+<Image
+  src="/images/deployment/troubleshoot/jwt-validation.png"
+  alt="jwt.io"
+  caption="jwt-validation"
```
[the remark this answers is not in the document]
Comment (MilanBariya) [reply openmetadata-docs/content/deployment/security/jwt-troubleshooting.md:47]: Yes it's correct.

**Review by nahuelverdugo — DISMISSED** [review]

File: openmetadata-docs/content/deployment/kubernetes/gke-troubleshooting.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+  src="/images/deployment/troubleshoot/permission-pod-events.png"
+  alt="permission-pod-events"
+  caption="Permission pod events"
+/>
+</div>
+
+Please validate:
+- all the prerequisites mentioned in this [section](/deployment/kubernetes/gke)
+- the configuration of `dags_pv_pvc.yml` file
+- `storageClassName` filed in YAML file
```
Comment (nahuelverdugo) [inline openmetadata-docs/content/deployment/kubernetes/gke-troubleshooting.md:29]:
```suggestion
- `storageClassName` field in YAML file
```
Field or filled?

File: openmetadata-docs/content/deployment/security/jwt-troubleshooting.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (nahuelverdugo) [reply openmetadata-docs/content/deployment/security/jwt-troubleshooting.md:47]: I think [this](https://github.com/open-metadata/OpenMetadata/commit/23c2fc09b66998b3b5e6f330ffeaf786e244b7e4) change got lost.

File: openmetadata-docs/content/deployment/security/jwt-troubleshooting.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (nahuelverdugo) [reply openmetadata-docs/content/deployment/security/jwt-troubleshooting.md:37]: I think [this](https://github.com/open-metadata/OpenMetadata/commit/23c2fc09b66998b3b5e6f330ffeaf786e244b7e4) change got lost.

File: openmetadata-docs/content/deployment/security/jwt-troubleshooting.md [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
```
Comment (nahuelverdugo) [inline openmetadata-docs/content/deployment/security/jwt-troubleshooting.md:47]:
```suggestion
  caption="jwt.io tool for validating JWT claims"
```

File: openmetadata-docs/content/deployment/security/jwt-troubleshooting.md [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+
+## Get JWT token from UI.
+
+First Open Open-Metadata UI than go to settings > Bots > Ingestion Bot
+
+<div className="w-100 flex justify-center">
+<Image
+  src="/images/deployment/troubleshoot/jwt-token.png"
+  alt="jwt-token"
+  caption="jwt-token"
```
Comment (nahuelverdugo) [inline openmetadata-docs/content/deployment/security/jwt-troubleshooting.md:37]:
```suggestion
  caption="JWT token in Openmetada UI"
```

File: openmetadata-docs/content/deployment/security/jwt-troubleshooting.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (MilanBariya) [reply openmetadata-docs/content/deployment/security/jwt-troubleshooting.md]: Done

File: openmetadata-docs/content/deployment/security/jwt-troubleshooting.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (MilanBariya) [reply openmetadata-docs/content/deployment/security/jwt-troubleshooting.md]: Done

**Review by nahuelverdugo — APPROVED** [review]

**Pull request merged**